File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var eraseOverlapIntervals = function(intervals) {
    intervals.sort((a,b) => a[1] - b[1])// sort by end time

    let count = 0
    let prevEnd = intervals[0][1]
    for(let i=1; i<intervals.length; i++){
        if(intervals[i][0] >= prevEnd){
            prevEnd = intervals[i][1]
        } else {
            count++
            prevEnd = Math.min(intervals[i][1], prevEnd) // remove later end
        }
    }
    return count
};

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: x[0])

        count = 0
        prevEnd = intervals[0][1]

        for i in range(1, len(intervals)):
            curr = intervals[i]
            if curr[0] >= prevEnd:
                prevEnd = curr[1]
            else:
                if prevEnd < curr[1]:
                    count += 1
                else:
                    count += 1
                    prevEnd = curr[1]
        
        return count
        